src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -114,20 +180,13 @@ public Task FindReferencesAsync(ISymbol symbol, CancellationToken cancellationTo
                 // which is why we do it in this loop and not inside the concurrent project processing that happens
                 // below.
                 await symbolSet.InheritanceCascadeAsync(currentProject, cancellationToken).ConfigureAwait(false);
-                allSymbols = symbolSet.GetAllSymbols();
+                var allSymbols = symbolSet.GetAllSymbols();
 
                 // Report any new symbols we've cascaded to to our caller.
                 await ReportGroupsAsync(allSymbols, cancellationToken).ConfigureAwait(false);
```

Comment:
I assume this can't be moved into the parallel part of the computation?

Reply:
let me noodle on that.  it would be clunkier.  Fundamentally, the inheritance cascade is serial.  The reporting of results it finds is... not that important.  but the reporting does need to happen *prior* to any symbols for a group being fired.